app/backend/approaches/chatreadretrieveread.py

```diff
@@ -48,7 +50,7 @@ def __init__(self, search_client: SearchClient, chatgpt_deployment: str, gpt_dep
         self.sourcepage_field = sourcepage_field
         self.content_field = content_field
 
-    def run(self, history: list[dict], overrides: dict) -> any:
+    def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]) -> Any:
```

Comment:
Similar to other comment, mypy doesn't like how this signature disagrees with Approach() signature.